File: MATLAB/schmittTrigger.m
function out=schmittTrigger(xi,outim1,schmittTriggerThresholdPerc, schmittTriggerOutLevel)
   %out = schmittTriggerOutLevel*(sign_local(xi + sign_local(outim1)*schmittTriggerThreshold));
   %out = schmittTriggerOutLevel*(sign(xi + sign(outim1)*schmittTriggerThreshold));
   %out = -schmittTriggerOutLevel*sign(xi - outim1*schmittTriggerThreshold);
   %out = -schmittTriggerOutLevel*sign(xi - outim1*schmittTriggerThresholdPerc);
   out = schmittTriggerOutLevel*sign(xi + outim1*schmittTriggerThresholdPerc);
end

File: ArrayTB/SchmittTrigger.cpp
#include "schmittTrigger.h"
float schmittTrigger(float xi, float outim1, float schmittTriggerTresholdPerc, float schmittTriggerOutLevel){
    float sig = xi+outim1*schmittTriggerTresholdPerc;
    int signum;
    if(sig < 0){
        signum = -1;
    }
    else if(sig == 0){
        signum = 0;
    }
    else{
        signum = 1;
    }

    return schmittTriggerOutLevel*signum;
}